tabular/src/autogluon/tabular/models/fasttext/fasttext_model.py

```diff
@@ -0,0 +1,158 @@
+__all__ = ["FastTextModel"]
+
+
+import logging
+import tempfile
+from collections import defaultdict
+
+import numpy as np
+import pandas as pd
+from pandas.api.types import is_numeric_dtype
+
+from autogluon.core.utils.loaders import load_pkl
+from autogluon.core.utils.savers import save_pkl
+
+from ...features.feature_metadata import FeatureMetadata
+from ...features.feature_metadata import R_CATEGORY, R_OBJECT,  S_TEXT
+
+from ...constants import BINARY, MULTICLASS
+from ..abstract.abstract_model import AbstractModel
+from .hyperparameters.parameters import get_param_baseline
+
+logging.basicConfig(
+    format="%(asctime)s: [%(funcName)s] %(message)s", level=logging.INFO
+)
+logger = logging.getLogger(__name__)
+
+
+def try_import_fasttext():
+    try:
+        import fasttext
+
+        _ = fasttext.__file__
+    except Exception:
+        raise ImportError('Import fasttext failed. Please run "pip install fasttext"')
+
+
+class FastTextModel(AbstractModel):
+    model_bin_file_name = "fasttext.bin"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        if self.problem_type not in (BINARY, MULTICLASS):
+            raise ValueError(
+                "FastText model only supports binary or multiclass classification"
+            )
+
+    def _set_default_params(self):
+        default_params = get_param_baseline()
+        for param, val in default_params.items():
+            self._set_default_param_value(param, val)
+
+    def _get_default_auxiliary_params(self) -> dict:
+        default_auxiliary_params = super()._get_default_auxiliary_params()
+        extra_auxiliary_params = dict(
+            get_features_kwargs=dict(
+                valid_raw_types=[R_CATEGORY, R_OBJECT, S_TEXT],
+            )
+        )
+        default_auxiliary_params.update(extra_auxiliary_params)
+        return default_auxiliary_params
+
+    def _fit(self, X_train, y_train, **kwargs):
+        try_import_fasttext()
+        import fasttext
+
+        X_train = self.preprocess(X_train)
+        logger.info("NLP features %s", self.features)
+
+        self.label_dtype = y_train.dtype
+        self.label_map = {label: f"__label__{i}" for i, label in enumerate(y_train.unique())}
+        self.label_inv_map = {v: k for k, v in self.label_map.items()}
+        idxs = np.random.permutation(list(range(len(X_train))))
+        with tempfile.NamedTemporaryFile(mode="w+t") as f:
+            logger.debug("generate training data")
+            for label, text in zip(y_train.iloc[idxs], (X_train[i] for i in idxs)):
+                f.write(f"{self.label_map[label]} {text}\n")
+            f.flush()
+            logger.debug("train FastText model")
+            self.model = fasttext.train_supervised(f.name, **self.params)
+            logger.debug("finish training FastText model")
+
+    def _preprocess(self, X: pd.DataFrame, **kwargs) -> list:
```

Comment:
Add `# TODO: move logic to self._preprocess_nonadaptive()`

This doesn't need to be implemented in this PR, but can serve as a reminder for future optimization to avoid redundant processing when bagging.